### DISCUSSION TITLE
Images in Cells

### DISCUSSION THREAD

**Discussion by MarkBaker** [PHPOffice/PhpSpreadsheet#3168] [General]
There is a common misconception that images can be placed in cells. This isn't the case; images (including charts) are in a layer that floats above the spreadsheet, like on a transparency, hiding the spreadsheet behind them, but not a part of the spreadsheet itself. Any positioning and sizing so that an image appears to be inside a cell has always been the responsibility of the developer, not of PhpSpreadsheet.

That is about to change, as Microsoft plan to implement a new Excel [`IMAGE()` function](https://insider.office.com/en-us/blog/insert-images-in-cells-with-the-image-function-in-excel), probably rolled out on release sometime in December. This doesn't allow images that are embedded in the file, but it is linked image urls.

I've just got access to this through the Beta channel; so I'll be experimenting with it over the next week or two, to see how it works internally, and to see if I can implement it.

It is likely to have some effect on existing functionality, such as autosizing columns. So it may take a bit longer to implement.

**Comment by luanmm** [comment]
Hello, @MarkBaker.

Anything evolved on this subject? I have a need to work with images, but the "embedded way" makes the resulting file too big.

I tried a little bit, rewriting Html reader class, to extract data from `<img />` tags and put into IMAGE() function. It works, but I couldn't figure out a way to output a file that seamlessy could open in Excel (specially the web version) - it works on Google Spreadsheets, by the way.

It would be great to work with this function, but in a way that the generated output could be opened in Excel without the need to run the "compatibility tool".

If you have any tips or findings regarding this, could you share?

Thanks in advance!

**Comment by oleibman** [comment]
Implemented with PR #4677. Closing.
